scripts/license_scanner/license_scanner.py

```diff
@@ -0,0 +1,128 @@
+
+    def _load_license_regex(self):
+        """Returns regex object with case-insensitive matching from the list of filename patterns."""
+        regex_patterns = []
+        for pattern in self.config_data['license_patterns']:
+            regex_patterns.append(fnmatch.translate(pattern))
+        return re.compile('|'.join(regex_patterns), re.IGNORECASE)
+
+    def scan(self, path=os.curdir):
+        """Scan directory tree for filenames matching license_regex.
+
+        :param path: Path of the directory to run scanner
+        :return: Package paths and their corresponding license file contents
+        :rtype: dict
+        """
+        licenses = 0
+        license_files = {}
+
+        for dirpath, dirnames, filenames in os.walk(path):
+            for file in filenames:
+                if self.license_regex.match(file):
+                    license_file_content = self._get_license_file_contents(os.path.join(dirpath, file))
+                    rel_dirpath = os.path.relpath(dirpath, path)  # Limit path inside scanned directory
+                    license_files[rel_dirpath] = license_file_content
+                    licenses += 1
+                    print(f'License file: {os.path.join(dirpath, file)}')
+
+            # Remove directories that should not be scanned
+            for dir in self.config_data['excluded_directories']:
+                if dir in dirnames:
+                    dirnames.remove(dir)
+        print(f'{licenses} license files found.')
+        return license_files
+
+    def _get_license_file_contents(self, filepath):
+        try:
+            with open(filepath, encoding='utf8') as f:
+                return f.read()
+        except UnicodeDecodeError:
+            print(f'Unable to read license file: {filepath}')
+            pass
+
+    def create_license_file(self, licenses, filepath='NOTICES.txt'):
+        """Creates file with all the provided license file contents.
+
+        :param licenses: Dict with package paths and their corresponding license file contents
+        :param filepath: Path to write the file
+       """
+        package_separator = '------------------------------------'
+        with open(filepath, 'w', encoding='utf8') as f:
+            for directory, license in licenses.items():
+                license_output = '\n\n'.join([
+                    f'{package_separator}',
+                    f'Package path: {directory}',
+                    'License:',
+                    f'{license}\n'
+                ])
+                f.write(license_output)
+        return None
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Script to run LicenseScanner and generate license file')
+    parser.add_argument('--config-file', '-c', help='Config file for LicenseScanner')
+    parser.add_argument('--license-file-path', '-l', help='Create license file in the provided path')
+    parser.add_argument('--scan-path', '-s', default=os.curdir, help='Path to scan')
```

Comment:
The default value is missing the parenthesis to invoke the method call. It should `os.curdir()`

Reply:
Also all three of these parameters should use the type of `pathlib.Path` to enforce a path type is being supplied

Reply:
Good call. I'll update the type for the parameters. 

os.curdir is the const string provided the OS for the cwd. You may be thinking of os.getcwd()

Reply:
Sorry about that, my mistake.